src/lib/mina_net2/bitswap_block.ml

```diff
@@ -0,0 +1,237 @@
+open Core_kernel
+
+let or_error_list_bind ls ~f =
+  let open Or_error.Let_syntax in
+  let rec loop = function
+    | [] ->
+        return []
+    | h :: t ->
+        let%bind r = f h in
+        let%map t' = loop t in
+        r :: t'
+  in
+  loop ls >>| List.concat
+
+type link = Blake2.t
+
+let link_size = Blake2.digest_size_in_bytes
+
+let absolute_max_links_per_block = Stdint.Uint16.(to_int max_int)
+
+type schema =
+  { total_blocks : int
```

Comment:
The field names here make this pretty hard to follow, there are a lot of 'max', 'link', and 'block'. Would it be possible to tweak the structure and names in this, perhaps to something like
```ocaml
{ num_link_only_blocks : int
; num_data_only_blocks : int
; num_links_in_link_and_data_block : int
; final_block_data_size : int
; num_bytes_per_block : int
; max_links_per_block : int }
```
(names could use bikeshedding, I'm not sure that I've captured the intention correctly..)

Reply:
I've refactored this to something close to what @mrmr1993, with some name changes for accuracy (eg `link_only_blocks` is a misnomer as link blocks still contain data in most cases). I do think that these names are a little confusing though, and propose that we consider moving to a different naming pattern that is clearer. I would consider `branch_blocks` and `leaf_blocks` to be more descriptive. Within that definition, there would be `n` `full_branch_blocks` and 0 or 1 `partial_branch_block` that has less links than the `max_link_per_branch_block`.

How do you two feel about that naming pattern @mrmr1993 and @georgeee?

Reply:
I like it, that seems much clearer to me

Reply:
I'm ok with this new naming pattern, let's use it